ambry-store/src/main/java/com/github/ambry/store/BlobStore.java

```diff
@@ -772,6 +775,183 @@ public void delete(List<MessageInfo> infosToDelete) throws StoreException {
     }
   }
 
+  @Override
+  public StoreBatchDeleteInfo batchDelete(List<MessageInfo> infosToDelete) throws StoreException {
```

Comment:
Can we look at breaking this function in smaller pieces?
Currently it stands at 180+ loc.